motr/io_req_fop.c

```diff
@@ -165,25 +165,32 @@ static void application_attribute_copy(struct m0_indexvec *rep_ivec,
 
 	rep_index	= m0_ivec_cursor_index(&rep_cursor);
 	ti_cob_index	= m0_ivec_cursor_index(&ti_cob_cursor);
+	ti_goff_index = m0_ivec_cursor_index(&ti_goff_cursor);
 
 	/* Move rep_cursor on unit boundary */
 	off = rep_index % unit_size;
 	if (off) {
-		if (m0_ivec_cursor_move(&rep_cursor, unit_size - off)) {
+		if (!m0_ivec_cursor_move(&rep_cursor, unit_size - off)) {
 			rep_index = m0_ivec_cursor_index(&rep_cursor);
 		}
 		else {
 			/** invalid cusror position */
-			M0_ASSERT(false);
+			return;
 		}
 	}
-	M0_ASSERT(ti_cob_index <= rep_index);
-
-	/* Move ti index to rep index */
-	if (ti_cob_index != rep_index) 	{
-		M0_ASSERT (m0_ivec_cursor_move(&ti_cob_cursor,  rep_index - ti_cob_index) && 
-		           m0_ivec_cursor_move(&ti_goff_cursor, rep_index - ti_cob_index));
+	off = ti_cob_index % unit_size;
+	if (off) {
```

Comment:
if (off != 0), please change this here and elsewhere.

Reply:
done